src/menu/cheat_load.c

```
@@ -0,0 +1,197 @@
+        path_free(path);
+        return CHEAT_LOAD_ERR_CHEAT_TOO_LARGE;
+    }
+
+    char *cheatsContent = NULL;
+    if((cheatsContent = malloc((cheatsLength + 1) * sizeof(char))) == NULL) {
+        path_free(path);
+        return CHEAT_LOAD_ERR_MALLOC_FAILED;
+    }
+    if(fread(cheatsContent, cheatsLength, 1, cheatsFile) != 1) {
+        path_free(path);
+        return CHEAT_LOAD_ERR_READ_FAILED;
```

cppcheck (2.10-2)

[error] Memory leak

(memleak)